### ISSUE TITLE
[feature request] nn.Identity

### ISSUE THREAD

**Issue by elbaro** [pytorch/pytorch#9160]
nn.Identity (similar to tf.identity or tf.no_op) would be useful as a placeholder.

## Use cases
it is common to parameterize building blocks (e.g. ResBlock depth, Bottleneck width, etc).
nn.Identity can fill in gaps to provide a consistent architecture.

## Code example
To simulate padding='SAME' in TF, nn.ZeroPad2d((0,1,0,1)) is conditionally inserted.

```py
def SamePadding(num_inputs, num_outputs, kernel_size, stride):
    if stride == 2:
        return nn.Sequential(
            nn.ConvTranspose2d(num_inputs, num_outputs, kernel_size, stride, bias=False,
                               padding=1),
            nn.ZeroPad2d((0, 0, 0, 0)),
            nn.BatchNorm2d(num_outputs),
            nn.ReLU()
        )
    else:
        return nn.Sequential(
            nn.ConvTranspose2d(num_inputs, num_outputs, kernel_size, stride, bias=False,
                               padding=2),
            nn.ZeroPad2d((0, 1, 0, 1)),
            nn.BatchNorm2d(num_outputs),
            nn.ReLU()
        )
```
By inserting `ZeroPad2d((0, 0, 0, 0))`, `nn.Sequential` is always of length 4, which makes the debugging and network surgery easier.

**Comment by ruotianluo** [comment]
You can use nn.Sequential() to simulate identity

**Comment by manuelsh** [comment]
You can also do a python function such as `lambda x: x`

**Comment by qualiaa** [comment]
Potential use-case: modifying a pre-defined network to easily remove layers.

    net = tv.models.alexnet(pretrained=True)
    net.classifier = Identity()

Replacing `Identity()` with `lambda x:x` yields an error. Sequential as currently implemented works fine, but it isn't at all clear from the documentation that this behaviour is guaranteed against future API changes.

**Comment by MilesCranmer** [comment]
nn.Sequential() is good for quick use. However if you want to easily exchange nn.Sequential as a placeholder for some module that takes arguments, you will encounter an error because nn.Sequential is argument-sensitive. Therefore the following is a good hack to get around this:

```python
class Identity(nn.Module):
    def __init__(self, *args, **kwargs):
        super().__init__()
    def forward(self, x):
        return x
```

Then, for example, you can write:
```python
batch_norm = nn.BatchNorm2d
if dont_use_batch_norm:
    batch_norm = Identity
```

Then in your network, you have:

```python
nn.Sequential(
    ...
    batch_norm(N, momentum=0.05),
    ...
)
```

and you can turn this into an identity with dont_use_batch_norm=True, and it won't trigger an error.

**Comment by MilesCranmer** [comment]
This PR adds an identity module: #19249.

**Comment by johntiger1** [comment]
> You can use nn.Sequential() to simulate identity

Not quite, since we can't swap out `nn.BatchNorm1d()` with `nn.Sequential()` directly. 

In particular, running your input x through nn.Sequential() returns none for me

**Comment by MilesCranmer** [comment]
FYI that PR has been merged. nn.Identity() is in master.

**Comment by johntiger1** [comment]
Thanks! Did not realize that. Here is a link to documentation for anyone in the future :) 
https://pytorch.org/docs/stable/nn.html#identity
